### ISSUE TITLE
relations between terms by ECL query

### ISSUE THREAD

**Issue by MahboobehJannesari** [IHTSDO/snowstorm#183]
Hello, 

I would like to know how can I find out is there any relation between two concepts or no? if there is, whats the relationship?
 how I do write the ECL query to find the relationship?

for example 
28275007 |Total body plethysmography (procedure)
255274008 |Pre-bronchodilation (qualifier value)|

**Comment by peterdutey** [comment]
This is an interesting question. I'm very interested in finding out from the pros how you would write the ECL query.
In the meantime I was wondering whether this might address another part of the question:

Those two concepts don't seem related
[https://browser.ihtsdotools.org/snowstorm/snomed-ct/MAIN/relationships?source=28275007&destination=255274008&offset=0&limit=50](https://browser.ihtsdotools.org/snowstorm/snomed-ct/MAIN/relationships?source=28275007&destination=255274008&offset=0&limit=50 )

However there is a relationship targeting 129266000 | Measurement - action (qualifier value) |
[https://browser.ihtsdotools.org/snowstorm/snomed-ct/MAIN/relationships?source=28275007&destination=129266000&offset=0&limit=50](https://browser.ihtsdotools.org/snowstorm/snomed-ct/MAIN/relationships?source=28275007&destination=129266000&offset=0&limit=50)

Hope this isn't completely off!

**Comment by kaicode** [comment]
I agree with @peterdutey, searching for relationship records is a great way to find direct relationships between two concepts. Using this technique the `type` field in the results tells us what type of relationship it is.

ECL can be used to test for the presence of relationships between specific concepts, or groups of concepts, but ECL can only return concepts.

ECL can also be used to test if there is a parent / child or ancestor / descendant relationship between two concepts. For example this ECL tests if ` 28275007 | Total body plethysmography (procedure) |` is a descendant of `64777005 | Physiologic measurement (procedure) |`:
```
<< 64777005 | Physiologic measurement (procedure) | AND 28275007 | Total body plethysmography (procedure) |
```
This returns `28275007 | Total body plethysmography (procedure) |` because it is a subtype of `64777005 | Physiologic measurement (procedure) | `.

This ECL tests to see if `255274008 |Pre-bronchodilation (qualifier value)|` or any subtype is used in any part of the model of `28275007 |Total body plethysmography (procedure)`
```
28275007 |Total body plethysmography (procedure) : * = << 255274008 |Pre-bronchodilation (qualifier value)|
```
This returns nothing because `28275007 |Total body plethysmography (procedure)` only uses the `260686004 |Method (attribute)|` attribute.

These ECL queries only give you a true / false answer, they do not tell us what type of relationship is it. If that information is required additional queries could be run to narrow down the attribute type or the whole set of relationships from `28275007 |Total body plethysmography (procedure)` could be loaded to be filtered in a script.
https://browser.ihtsdotools.org/snowstorm/snomed-ct/MAIN/2020-07-31/relationships?source=28275007&active=true

In your application be sure to use the latest version branch (e.g. `MAIN/2020-07-31`) which you can find on the [codesystem record](https://browser.ihtsdotools.org/snowstorm/snomed-ct/codesystems) rather than `MAIN` because `MAIN` is used for the daily build of the current authoring cycle. It contains unpublished content which is subject to change, including concept deletion.

I hope that helps.

A possible enhancement to the relationship search would be to allow ECL in the `source`, `type` and `destination` fields.

**Comment by MahboobehJannesari** [comment]
Thanks for your comments.

This query support all relation between synonym of source and target or no?
28275007 |Total body plethysmography (procedure) : * = << 255274008 |Pre-bronchodilation (qualifier value)|

I mean is it possible there is relation between synonyms of source and synonyms of target?

**Comment by kaicode** [comment]
In snomed a synonym is a type of description, is that what you mean? There can be multiple synonym descriptions which belong to the same concept.
ECL works on concepts, not specific descriptions. So you could say that if something applies to a concept then it applies to all the active descriptions of that concept.
In my examples I have included the FSN description only for readability. The following ECL statements would work just the same:
```
28275007 |Total body plethysmography (procedure) | : * = << 255274008 |Pre-bronchodilation (qualifier value)|
```
```
28275007 : * = << 255274008
```
I've just noticed that in my previous example I missed a pipe after the description which would have caused unpredictable results. I've corrected that now.
Does that answer your question?

**Comment by MahboobehJannesari** [comment]
yes.
Thank you
